umi_tools/umi_methods.py

```diff
@@ -95,11 +95,11 @@ def write(self, read, unique_id=None, umi=None, unmapped=False):
             self.outfile.write(read)
             return
 
-        if not self.chrom == read.reference_id:
+        if not self.chrom == read.reference_name:
             self.write_mates()
-            self.chrom = read.reference_id
+            self.chrom = read.reference_name
 
-        key = read.query_name, read.next_reference_id, read.next_reference_start
+        key = read.query_name, read.reference_name, read.next_reference_start
```

Comment:
This needs to be:

`key = read.query_name, read.next_reference_name, read.next_reference_start`